functions/types.ts

```diff
@@ -1,7 +1,10 @@
 import { EventContext, KVNamespace } from "@cloudflare/workers-types";
+import { Request, IncomingRequestCfProperties } from "@cloudflare/workers-types";
 
 export interface Env {
   userToReferral: KVNamespace;
 }
 
-export type Context = EventContext<Env, string, Record<string, string>>;
\ No newline at end of file
+export type Context = EventContext<Env, string, Record<string, string>>;
+
+export type CustomRequest = Request<unknown, IncomingRequestCfProperties<unknown>>;
```

Comment:
are you sure this is needed? I've never needed to use it, usually just `Request`

Reply:
the type of the request I get here: `const { request, env } = ctx;` is the `Request<unknown, IncomingRequestCfProperties<unknown>>`, I tried to type cast the variable `request` to `Request`, but it was not successful, I think this strange request is specific to CloudFlare Workers.

Reply:
I have just tried locally with `Request` from `@cloudflare/workers-types` and it works normally

Reply:
Oh sorry, probably when I tried to use `Request` the import got it from another lib, and it didnt work, so I end up using that other one, I will update to use the `Request` from `@cloudflare/workers-types`